tools/java_api/java_test/ValueTest.java

```diff
@@ -0,0 +1,857 @@
+        try{
+            TestHelper.cleanup();
+            db.destory();
+            conn.destory();
+        }catch(AssertionError e) {
+            fail("destroyDBandConn failed: ");
+            System.out.println(e.toString());
+        }
+        System.out.println("Data cleaned up");
+    }
+
+    @Test
+    void ValueCreateNull() {
+        KuzuValue value = KuzuValue.createNull();
+        assertFalse(value.isOwnedByCPP());
+        assertEquals(value.getDataType().getID(), KuzuDataTypeID.ANY);
+        value.destroy();
+
+        System.out.println("ValueCreateNull passed");
+    }
+
+    @Test
+    void ValueCreateNullWithDatatype() {
+        KuzuDataType type = new KuzuDataType(KuzuDataTypeID.INT64, null, 0);
+        KuzuValue value = KuzuValue.createNullWithDataType(type);
+        assertFalse(value.isOwnedByCPP());
+        type.destory();
+
+        assertEquals(value.getDataType().getID(), KuzuDataTypeID.INT64);
+        value.destroy();
+
+        System.out.println("ValueCreateNullWithDatatype passed");
+    }
+
+    @Test
+    void ValueIsNull() {
+        KuzuValue value = new KuzuValue(123L);
+        assertFalse(value.isOwnedByCPP());
+        assertFalse(value.isNull());
+        value.destroy();
+
+        value = KuzuValue.createNull();
+        assertTrue(value.isNull());
+        value.destroy();
+
+        KuzuDataType type = new KuzuDataType(KuzuDataTypeID.INT64, null, 0);
+        value = KuzuValue.createNullWithDataType(type);
+        assertTrue(value.isNull());
+        type.destory();
+        value.destroy();
+
+        System.out.println("ValueIsNull passed");
+    }
+
+    @Test
+    void ValueSetNull() {
+        KuzuValue value = new KuzuValue(123L);
+        assertFalse(value.isOwnedByCPP());
+        assertFalse(value.isNull());
+
+        value.setNull(true);
+        assertTrue(value.isNull());
+
+        value.setNull(false);
+        assertFalse(value.isNull());
+        value.destroy();
+
+        System.out.println("ValueSetNull passed");
+    }
+
+    @Test
+    void ValueCreateDefault() {
+        KuzuDataType type = new KuzuDataType(KuzuDataTypeID.INT64, null, 0);
+        KuzuValue value = KuzuValue.createDefault(type);
+        assertFalse(value.isOwnedByCPP());
+        type.destory();
+
+        assertFalse(value.isNull());
+        assertEquals(value.getDataType().getID(), KuzuDataTypeID.INT64);
+        assertTrue(value.getValue().equals(0L));
+        value.destroy();
+
+        type = new KuzuDataType(KuzuDataTypeID.STRING, null, 0);
+        value = KuzuValue.createDefault(type);
+        assertFalse(value.isOwnedByCPP());
+        type.destory();
+
+        assertFalse(value.isNull());
+        assertEquals(value.getDataType().getID(), KuzuDataTypeID.STRING);
+        assertTrue(value.getValue().equals(""));
+        value.destroy();
+
+        System.out.println("ValueCreateDefault passed");
+    }
+
+    @Test
+    void ValueCreate() {
```

Comment:
Split this long test case into multiple shorter test cases for each type, so it is easier to maintain and debug when a test case fail.